### ISSUE TITLE
Storage unit becomes full for no reason

### ISSUE THREAD

**Issue by GabFran** [xenondevs/Nova#505]
### Agreements

- [X] I am using an official build of Nova.
- [X] I am running the latest version of Nova.
- [X] I am using a [server software supported by Nova](https://docs.xen.cx/nova/admin/compatibility/#compatible-server-software)
- [X] I am running the latest version of my server software.
- [X] I have checked for similar issues.

### Check for plugin incompatibilites

There are no other plugins running on my server.

### Plugin Version, Server Software, Server Version

Nova: 0.17-alpha.30; Paper: 1.21.1-119-master@7cd4f2c; MC: 1.21.1

### Nova Addons and Versions

Simple-Upgrades v1.4-alpha.2, Machines v0.4.6-alpha.13, Logistics v0.2.9-alpha.5

### Expected behavior

Storage units aren't supposed to randomly fill up to the 32-bit integer limit

### Observed/Actual behavior

Sometimes, (apparently only after server restarts), storage units fill up to their maximum capacity (32-bit integer limit)

![image](https://github.com/user-attachments/assets/975e59dc-e5a3-46d6-919b-017e9a1810c9)

### Steps to reproduce

1. I have not been able to consistently reproduce this bug, but it has only happened when storage units which contain items are outputting to an adjacent inventory, such as a chest or barrel.
2. It has happened after a server restart both times.
3. The storage unit must contain at least one item.
4. The console is spammed with an apparent loop for the affected storage units (it cites the items which are being multiplied multiple times in the error)

### Server log

https://api.mclo.gs/1/raw/8bUh69a

### Errors

```log
[16:33:07] [DefaultDispatcher-worker-3/ERROR]: [Nova] An exception occurred trying to tick xyz.xenondevs.nova.world.block.tileentity.network.type.item.ItemNetworkGroup@8684125 in cluster NetworkCluster(uuid=deabb8f2-273e-4a7b-ab19-e77d134fd155, networks=[ItemNetwork(nodes={BlockPos(world=world, x=4, y=70, z=113)=MutableNetworkNodeConnection(node=StorageUnit(blockState=logistics:storage_unit[], pos=BlockPos(world=world, x=4, y=70, z=113), data={
  "itemHolder": {
    "channels": {NORTH=0, EAST=0, SOUTH=0, WEST=0, UP=0, DOWN=0}
    "connectionConfig": {NORTH=NONE, EAST=NONE, SOUTH=NONE, WEST=EXTRACT, UP=NONE, DOWN=INSERT}
    "containerConfig": {NORTH=045a8457-aa96-4feb-878a-39c069cbbde4, EAST=045a8457-aa96-4feb-878a-39c069cbbde4, SOUTH=045a8457-aa96-4feb-878a-39c069cbbde4, WEST=045a8457-aa96-4feb-878a-39c069cbbde4, UP=045a8457-aa96-4feb-878a-39c069cbbde4, DOWN=045a8457-aa96-4feb-878a-39c069cbbde4}
    "extractFilters": {}
    "extractPriorities": {NORTH=50, EAST=50, SOUTH=50, WEST=50, UP=50, DOWN=50}
    "insertFilters": {DOWN=xyz.xenondevs.nova.addon.logistics.item.itemfilter.TypeItemFilter@2b71246a}
    "insertPriorities": {NORTH=50, EAST=50, SOUTH=50, WEST=50, UP=50, DOWN=50}
  }
  "ownerUuid": (serialized) 013e67b1b3f6e94aba8c6a26f54cb75c53
  "persistent": {
    "amount": 0
    "type": ItemStack{AIR x 0}
  }
```

**Comment by NichtStudioCode** [comment]
Is this a new issue with 0.17-alpha.30 or did it also happen previously?

**Comment by NichtStudioCode** [comment]
I have also pushed a fix for something that may relate to this in 0.17-alpha.31, please tell me if you can still observe it under that version

**Comment by GabFran** [comment]
This has happened in alpha 27 aswell

**Comment by GabFran** [comment]
> I have also pushed a fix for something that may relate to this in 0.17-alpha.31, please tell me if you can still observe it under that version

Will keep a look out for that, cheers.

**Comment by GabFran** [comment]
Haven't seen this happening ever since I posted this, so I will assume it is fixed.
